src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -686,7 +686,10 @@ roundUpToPower2 x =
         z3 = z2 .|. z2 `shiftR` 4
         z4 = z3 .|. z3 `shiftR` 8
         z5 = z4 .|. z4 `shiftR` 16
-        in z5 + 1
+        z6 = if (maxBound :: Int) == 9223372036854775807
```

Comment:
Its not a good idea to relay on maxBound. It also places a branch and comparison in this code which makes it less efficient. Let's do it statically, [see this for example](https://hackage.haskell.org/package/statistics-0.15.2.0/docs/src/Statistics.Function.html#nextHighestPowerOfTwo).